lms/djangoapps/badges/models.py

```diff
@@ -0,0 +1,328 @@
+"""
+Database models for the badges app
+"""
+from importlib import import_module
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from lazy import lazy
+from model_utils.models import TimeStampedModel
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from config_models.models import ConfigurationModel
+from xmodule.modulestore.django import modulestore
+from xmodule_django.models import CourseKeyField
+from jsonfield import JSONField
+
+
+def validate_badge_image(image):
+    """
+    Validates that a particular image is small enough, of the right type, and square to be a badge.
+    """
+    if image.width != image.height:
+        raise ValidationError(_(u"The badge image must be square."))
+    if not image.size < (250 * 1024):
+        raise ValidationError(_(u"The badge image file size must be less than 250KB."))
+
+
+def validate_lowercase(string):
+    """
+    Validates that a string is lowercase.
+    """
+    if not string.islower():
+        raise ValidationError(_(u"This value must be all lowercase."))
+
+
+class CourseBadgesDisabledError(Exception):
+    """
+    Exception raised when Course Badges aren't enabled, but an attempt to fetch one is made anyway.
+    """
+
+
+class BadgeClass(models.Model):
+    """
+    Specifies a badge class to be registered with a backend.
+    """
+    slug = models.SlugField(max_length=255, validators=[validate_lowercase])
+    issuing_component = models.SlugField(max_length=50, default='', blank=True, validators=[validate_lowercase])
+    display_name = models.CharField(max_length=255)
+    course_id = CourseKeyField(max_length=255, blank=True, default=None)
+    description = models.TextField()
+    criteria = models.TextField()
+    # Mode a badge was awarded for. Included for legacy/migration purposes.
+    mode = models.CharField(max_length=100, default='', blank=True)
+    image = models.ImageField(upload_to='badge_classes', validators=[validate_badge_image])
+
+    def __unicode__(self):
+        return u"<Badge '{slug}' for '{issuing_component}'>".format(
+            slug=self.slug, issuing_component=self.issuing_component
+        )
+
+    @classmethod
+    def get_badge_class(
```

Comment:
I'm fine leaving as-is -- can we add this helpful context to the docstring or as a comment?